gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,242 @@
+"""Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+Authors: Travis Driver
+"""
+
+import os
+from pathlib import Path
+from typing import Optional, Dict, Tuple, List
+
+from gtsam import Cal3Bundler, Pose3, Rot3, Point3, SfmTrack
+import trimesh
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+from gtsfm.utils.read_write_model import read_model
+from gtsfm.utils.read_write_model import Camera as ColmapCamera
+from gtsfm.utils.read_write_model import Image as ColmapImage
+from gtsfm.utils.read_write_model import Point3D as ColmapPoint3D
+
+
+logger = logger_utils.get_logger()
+
+
+class AstroNetLoader(LoaderBase):
+    """Simple loader class that reads an AstroNet data segment.
+
+    Refs:
+    - https://github.com/travisdriver/astronet
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        gt_scene_mesh_path: str = None,
+        use_gt_intrinsics: bool = True,
+        use_gt_extrinsics: bool = True,
+        use_gt_sfmtracks: bool = False,
+        max_frame_lookahead: int = 2,
+    ) -> None:
+        """Initialize loader from a specified segment directory (data_dir) on disk.
+
+        <data_dir>/
+             ├── images/: undistorted grayscale images
+             ├── cameras.bin: camera calibrations (see https://colmap.github.io/format.html#cameras-txt)
+             ├── images.bin: 3D poses and 2D tracks (see https://colmap.github.io/format.html#images-txt)
+             └── points3D.bin: 3D tracks (see https://colmap.github.io/format.html#points3d-txt)
+
+
+        Args:
+            data_dir: path to directory containing the COLMAP-formatted data: cameras.bin, images.bin, and points3D.bin
+            gt_scene_mesh_path (optional): path to file of target small body surface mesh.
+                Note: vertex size mismath observed when reading in from OBJ format. Prefer PLY.
+            use_gt_intrinsics: whether to use ground truth intrinsics. If calibration is
```

Comment:
do the astronet files have EXIF?

Reply:
No they do not.

Reply:
got it. how are intrinsics estimated if `use_gt_intrinsics` is set to false then?

Reply:
I guess I could have it require that ground truth intrinsics are provided.

Is the case for the intrinsics being `None` is handled anywhere? The output of `get_intrinsics_from_exif()` is `Optional[Cal3Bundler]`, but it seems the ground truth intrinsics (or a good estimate) are required.

Reply:
@johnwlambert Let me know how you think this should be handled

Reply:
If we do not have exif, we can approximate the intrinsics from image size? We might have the code somewhere.

I do not think we should make intrinsics as compulsory.

Reply:
The main issue would be approximating the focal length, especially for this case as the focal length for camera onboard spacecraft are usually very large (10,000 - 100,000 pixels).

Reply:
I don't see that issue with making it required just for this specific loader, as the ground truth intrinsics are always available.

Reply:
Made calibrations required for now. Opened handling of no intrinsics as an issue [#265](https://github.com/borglab/gtsfm/issues/265).